### PR TITLE
[onert/odc] Auto-compilation. Tests

### PR DESCRIPTION
**Pull request by Torrero**
This PR introduces tests for `odc:auto-compilation`. 

For [Issue]( https://github.com/Samsung/ONE/issues/13288). 
From [Draft](https://github.com/Samsung/ONE/pull/13530).

ONE-DCO-1.0-Signed-off-by: Evgenii Maltsev e.maltsev@samsung.com

### CODE REVIEW

File: tests/nnfw_api/CMakeLists.txt [hunk 1]
```diff
@@ -65,3 +65,7 @@ install(DIRECTORY ${NNPACKAGE_MODEL_DIR} DESTINATION ${NNPACKAGE_INSTALL_TARGET}
 # Install nnpackage test model (mobilenet)
 set(NNPACKAGE_MODEL_DIR ${NNAS_PROJECT_SOURCE_DIR}/runtime/contrib/TFLiteSharp/TFLiteTestApp/res/)
 install(DIRECTORY ${NNPACKAGE_MODEL_DIR} DESTINATION ${NNPACKAGE_INSTALL_TARGET}/mobilenet_v1_1.0_224)
+
+# Install nnpackage test model (conv2d)
+set(NNPACKAGE_MODEL_DIR ${NNAS_PROJECT_SOURCE_DIR}/runtime/contrib/autocompilation/res/)
+install(DIRECTORY ${NNPACKAGE_MODEL_DIR} DESTINATION ${NNPACKAGE_INSTALL_TARGET}/conv2d)
```
Comment (hseok-oh) [inline tests/nnfw_api/CMakeLists.txt:71]: Using `${NNAS_PROJECT_SOURCE_DIR}/runtime/contrib/TFLiteSharp/TFLiteTestApp/res/` is workaround to use large size model - mobilenet. `TFLiteSharp` was introduced as C# API PoC project. And all directories under `contrib` is not maintained.

Please use different place. (ex. `tests/nnfw_api/res`)

Comment (Torrero) [reply tests/nnfw_api/CMakeLists.txt]: @hseok-oh Thank you for explanation, 
I changed the location for the model.

**Review by hseok-oh — APPROVED** [review]

**Review by stamalakhov — APPROVED** [review]
LGTM! Thank you!

File: tests/nnfw_api/src/OdcAutoCompilation.test.cc [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  std::remove(quantized_model_name.c_str());
+  std::remove(compiled_model_name.c_str());
+
+  // setup ODC parameters
+  nnfw_set_quantized_model_path(session, quantized_model_name.c_str());
+  nnfw_set_quantization_type(session, NNFW_QUANTIZE_TYPE::NNFW_QUANTIZE_TYPE_U8_ASYM);
+
+  nnfw_set_codegen_model_path(session, compiled_model_name.c_str());
+
+  const int RUNS_COUNT_FOR_QUANTIZATION = input_tensors.size();
```
Comment (SlavikMIPT) [inline tests/nnfw_api/src/OdcAutoCompilation.test.cc:84]:
```suggestion
  const auto RUNS_COUNT_FOR_QUANTIZATION = input_tensors.size();
```

File: tests/nnfw_api/src/OdcAutoCompilation.test.cc [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  nnfw_create_session(&session);
+
+  nnfw_load_model_from_modelfile(session, (model_path + ".circle").c_str());
+  nnfw_set_available_backends(session, "cpu");
+  nnfw_prepare(session);
+
+  // Delete minmax file
+  nnfw_odc_delete_minmax_file(session);
+
+  const int RUNS_COUNT_FOR_QUANTIZATION = 1;
```
Comment (SlavikMIPT) [inline tests/nnfw_api/src/OdcAutoCompilation.test.cc:181]:
```suggestion
  const auto RUNS_COUNT_FOR_QUANTIZATION = 1;
```

**Review by SlavikMIPT — APPROVED** [review]
LGTM (optional changes in suggestions)

**Comment by hseok-oh** [comment]
We may updates all tests under `tests/nnfw_api/` to use @SlavikMIPT 's suggestion.

**Pull request merged**